File: Leetcode/2966 - Divide Array Into Arrays With Max Difference/solution.py
def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
    nums.sort()
    res_arr = []

    for i in range(2, len(nums), 3):
        if (nums[i] - nums[i - 2]) > k:
            return []
        else:
            res_arr.append(nums[i-2 : i+1])

    return res_arr

File: Leetcode/2966 - Divide Array Into Arrays With Max Difference/solution.js
var divideArray = function (nums, k) {
    let resArr = [];
    nums.sort((a, b) => a - b);
    for (let i = 2; i < nums.length; i += 3) {
        if (nums[i] - nums[i - 2] > k) return [];
        resArr.push(nums.slice(i - 2, i + 1));
    }
    return resArr;
};